mlflow/utils/validation.py

```diff
@@ -345,6 +345,19 @@ def _validate_experiment_name(experiment_name):
         )
 
 
+def _validate_experiment_id_type(experiment_id):
+    """
+    Check that a user-provided experiment_id is either a string, int, or None and raise an
+    exception if it isn't.
+    """
+    if experiment_id is not None and not isinstance(experiment_id, (str, int)):
```

Comment:
```suggestion
    if not isinstance(experiment_id, (str, int, type(None))):
```

Is this simpler?

Reply:
Never mind this comment. I think `experiment_id is not None` is more straightforward than `isinstance(x, type(None))`